File: leetcode/string/929.unique-email-addresses.py
class Solution:
    def isValid(self, email):
        string = email.split('@')
        string[0] = string[0].split('+')[0]
        string[0] = string[0].replace('.', '')
        return string[0] + '@' + string[1]

    def numUniqueEmails(self, emails: List[str]) -> int:
        for i in range(len(emails)):
            s = emails.pop(0)
            s = self.isValid(s)
            if s not in emails:
                emails.append(s)
        return len(emails)

File: leetcodeJS/string/929.unique-email-addresses.js
/**
 * @param {string[]} emails
 * @return {number}
 */

const formEmail = (email) => {
  let arr = email.split("@");
  let front = arr[0].split("+")[0];
  front = front.split(".").join("");
  return front + "@" + arr[1];
};

var numUniqueEmails = function (emails) {
  let solution = [];
  for (i of emails) {
    let email = formEmail(i);
    if (!solution.includes(email)) solution.push(email);
  }
  return solution.length;
};
